src/XrdTpc/XrdTpcMultistream.cc

```diff
@@ -281,6 +281,11 @@ int TPCHandler::RunCurlWithStreamsImpl(XrdHttpExtReq &req, State &state,
         curl_handles.emplace_back(handles.back()->GetHandle());
     }
 
+    // Notify the packet marking manager that the transfer will start after this point
+    if(rec.pmarkManager) {
+      rec.pmarkManager->setTransferWillStart(&req);
```

Comment:
This feels a bit awkward here, maybe a `startTransfer` in `TPCLogRecord` makes more sense, then it'd look like just `rec.startTransfer(&req)` here, and you hide some of the internals of `TPCLogRecord` for external callers. Same for other places where such indirections appear, and `if(rec.pmarkManager)` is needed.

Reply:
Yes, renaming setTransferWillStart :)